src/snowflake/snowpark/_internal/debug_utils.py

```diff
@@ -249,21 +277,273 @@ def get_python_source_from_sql_error(top_plan: "SnowflakePlan", error_msg: str)
+
+        nodes, children, root_nodes = self.build_operator_tree(stats_result)
+        file_handle = None
+        if output_file:
+            mode = "a" if append_mode else "w"
+            file_handle = open(output_file, mode, encoding="utf-8")
+
+        try:
+            self._write_output(f"\n=== Analyzing Query {query_id} ===", file_handle)
+            self._write_output(f"\n{'='*80}", file_handle)
+            self._write_output("QUERY OPERATOR TREE", file_handle)
+            self._write_output(f"{'='*80}", file_handle)
+
+            root_list = sorted(list(root_nodes))
+            for i, root_id in enumerate(root_list):
+                is_last_root = i == len(root_list) - 1
+                self.print_operator_tree(
+                    nodes, children, root_id, "", is_last_root, file=file_handle
+                )
+
+            self._write_output(f"\n{'='*160}", file_handle)
+            self._write_output("DETAILED OPERATOR STATISTICS", file_handle)
+            self._write_output(f"{'='*160}", file_handle)
+            self._write_output(
+                f"{'Operator':<15} {'Type':<15} {'Input Rows':<12} {'Output Rows':<12} {'Row Multiple':<12} {'Overall %':<12} {'Attributes':<50}",
+                file_handle,
+            )
+            self._write_output(f"{'='*160}", file_handle)
+
+            total_input = 0
+            total_output = 0
+
+            for row in stats_result:
+                operator_id = (
+                    row["OPERATOR_ID"]
+                    if "OPERATOR_ID" in row and row["OPERATOR_ID"] is not None
+                    else 0
+                )
+                operator_type = (
+                    row["OPERATOR_TYPE"]
+                    if "OPERATOR_TYPE" in row and row["OPERATOR_TYPE"] is not None
+                    else "N/A"
+                )
+                input_rows = (
+                    row["INPUT_ROWS"]
+                    if "INPUT_ROWS" in row and row["INPUT_ROWS"] is not None
+                    else 0
+                )
+                output_rows = (
+                    row["OUTPUT_ROWS"]
+                    if "OUTPUT_ROWS" in row and row["OUTPUT_ROWS"] is not None
+                    else 0
+                )
+                row_multiple = (
+                    row["ROW_MULTIPLE"]
+                    if "ROW_MULTIPLE" in row and row["ROW_MULTIPLE"] is not None
+                    else 0
+                )
+                exec_time = (
+                    row["OVERALL_PERCENTAGE"]
+                    if "OVERALL_PERCENTAGE" in row
+                    and row["OVERALL_PERCENTAGE"] is not None
+                    else 0
+                )
+                operator_attrs = (
+                    row["OPERATOR_ATTRIBUTES"]
+                    if "OPERATOR_ATTRIBUTES" in row
+                    and row["OPERATOR_ATTRIBUTES"] is not None
+                    else "N/A"
+                )
```

Comment:
this part seems duplicated from above. can this be extracted out into another method?